pkg/operator/controller/controller_service_monitor_test.go

```diff
@@ -0,0 +1,64 @@
+package controller
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestDNSServiceMonitorChanged(t *testing.T) {
+	testCases := []struct {
+		description string
+		mutate      func(*unstructured.Unstructured)
+		expect      bool
+	}{
+		{
+			description: "if nothing changes",
+			mutate:      func(_ *unstructured.Unstructured) {},
+			expect:      false,
+		},
+		{
+			description: "if spec.endpoints.scheme changes",
+			mutate: func(serviceMonitor *unstructured.Unstructured) {
+				serviceMonitor.Object["spec"] = map[string]interface{}{
+					"selector": map[string]interface{}{},
+					"endpoints": []interface{}{
+						map[string]interface{}{
+							"scheme": "http",
```

Comment:
Instead of overriding `spec`, is it feasible to update just this field?  If it would be a hassle to do so, this is fine, but it does mean that we risk false positives if anything else in the definition of `original` below differs from the literal value here.

Reply:
This is cumbersome, although I do see the risk of false positives. Maybe this is something that could be corrected with the potential switch to the prometheus operator's servicemonitor type?